### PR TITLE
Feature/aut 22/add ckeditor to inline choice

### PR DESCRIPTION
**Pull request by pnal**
https://oat-sa.atlassian.net/browse/AUT-22

related fe branch: https://github.com/oat-sa/tao-item-runner-qti-fe/tree/feature/AUT-22/add-ckeditor-to-inline-choice

**How to test**
- Create inline choice interaction
- Try to add different markup and math to the choices
- Add new / delete choices
- Show, set the response
- Save item
- Check preview

Env: http://test-anastasia.playground.kitchen.it.taocloud.org:44451/

### CODE REVIEW

**Comment by AnastasiaArcadia** [comment]
There's an issue with rendering option in response mode, I'm checking it on the FE side

**Review by bziondik — CHANGES_REQUESTED** [review]
1. Have issue with addChoice button
image

2. Math in inline choice is not rendered properly in item preview. Issue with item runner?Should be a) fixed in scope of this ticket or b) should be created separate ticket and this PR should be merged in working branch not develop.
image

3. Can't preview test
image


4. Can't publish test
image

File: views/js/qtiCreator/renderers/choices/InlineChoice.js [hunk 1]
```diff
@@ -1 +1,37 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```
[the remark this answers is not in the document]
Comment (pnal) [reply views/js/qtiCreator/renderers/choices/InlineChoice.js:16]: > 1. Have issue with addChoice button

Is it FE issue?
> 2. Math in inline choice is not rendered properly in item preview. Issue with item runner?Should be a) fixed in scope of this ticket or b) should be created separate ticket and this PR should be merged in working branch not develop.

We have a separate ticket for TR, so I changed the destination branch so this will not be merged to develop.
> 3. Can't preview test
> 4. Can't publish test

To fix the rendering we need to add changes in QTI-SDK cause now it supports only text variable for inline choice options https://github.com/oat-sa/qti-sdk/blob/legacy/qtism/data/storage/xml/marshalling/InlineChoiceMarshaller.php#L57. I fixed this on my local environment by replacing `TextOrVariableCollection` with `FlowStaticCollection`  for `inlineChoice` content type. But we need to ask @julien-sebire is it possible to add such changes in QTI-SDK and how we may release such breaking change.

[the remark this answers is not in the document]
Comment (AnastasiaArcadia) [reply views/js/qtiCreator/renderers/choices/InlineChoice.js:16]: > Is it FE issue?

Yes, probably, I'll take a look

[the remark this answers is not in the document]
Comment (AnastasiaArcadia) [reply views/js/qtiCreator/renderers/choices/InlineChoice.js]: Now should be fixed

**Comment by pnal** [comment]
Created [draft PR](https://github.com/oat-sa/qti-sdk/pull/287/files) for QTI-SDK which reflects kind of changes needed for support this feature.

File: views/js/qtiCreator/model/choices/InlineChoice.js [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+], function(_, editable, Choice){
     "use strict";
     var methods = {};
     _.extend(methods, editable);
     _.extend(methods, {
         getDefaultAttributes : function(){
             return {
-                'fixed' : false,
-                'showHide' : 'show'
+                fixed : false,
```
Comment (mike-ionut-mihai-sandu-tao) [inline views/js/qtiCreator/model/choices/InlineChoice.js:12]: why did you remove the quotes ?

File: model/qti/choice/InlineChoice.php [hunk 3]
```diff
@@ -15,26 +15,27 @@
...
  * @package taoQTI
-
  */
-class InlineChoice extends TextVariableChoice
+class InlineChoice extends ContainerChoice
 {
+    public function getValidElementTypes(): array
+    {
+        return [
+            'oat\\taoQtiItem\\model\\qti\\Math'
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/qti/choice/InlineChoice.php:36]: maybe let's use the Math::class

File: views/js/qtiCreator/widgets/interactions/inlineChoiceInteraction/Widget.js [hunk 4]
```diff
@@ -8,42 +8,58 @@ define([
...
 
        //remove toolbar title, because it is too large
        this.$container.find('.tlb-title').remove();
     };
 
-    InlineChoiceInteractionWidget.renderChoice = function(choice, shuffleChoice){
+    InlineChoiceInteractionWidget.renderChoice = function(choice, shuffleChoice) {
 
-        var tplData = {
+        const interaction = this.element;
```
Comment (mike-ionut-mihai-sandu-tao) [inline views/js/qtiCreator/widgets/interactions/inlineChoiceInteraction/Widget.js:32]: eslint, prettier ?

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: views/js/qtiCreator/model/choices/InlineChoice.js (hunk 2 above)
Comment (AnastasiaArcadia) [reply views/js/qtiCreator/model/choices/InlineChoice.js]: Result of prettier

**Review by AnastasiaArcadia — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not master

**Review by Silvia-loza — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/interactions/inlineChoiceInteraction/Widget.js [hunk 5]
```diff
@@ -6,84 +6,93 @@ define([
...
 
-    InlineChoiceInteractionWidget.renderChoice = function(choice, shuffleChoice){
+    InlineChoiceInteractionWidget.renderChoice = function (choice, shuffleChoice) {
+        const interaction = this.element;
+        const interactionData = { interaction };
+
+        const widget = choice.metaData.widget;
+        let container = choice.getBody();
+        if (widget) {
+            container = widget.element.getBody();
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/interactions/inlineChoiceInteraction/Widget.js:35]: Is widget.element === choice ?

Comment (AnastasiaArcadia) [reply views/js/qtiCreator/widgets/interactions/inlineChoiceInteraction/Widget.js]: Yes, you are right

**Review by bziondik — COMMENTED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/choices/inlineChoice/states/Question.js [hunk 6]
```diff
@@ -25,40 +31,61 @@ define([
...
-
-                //update placeholder
-                _widget.$original.width($(this).width());
-
-            }, 200)).on('focus.qti-widget', function(){
+    ChoiceStateQuestion.prototype.buildEditor = function () {
+        const _widget = this.widget,
+            container = _widget.element.getBody(),
+            $editable = _widget.$container.find('.editable-content'),
+            $editableContainer = _widget.$container.find('.editable-container');
```
[the remark this answers is not in the document]
Comment (bziondik) [reply views/js/qtiCreator/widgets/choices/inlineChoice/states/Question.js:38]: sorry, no issue

**Review by bziondik — APPROVED** [review]
